### ISSUE TITLE
Mongod does not cleanup

### ISSUE THREAD

**Issue by bence-takacs** [flapdoodle-oss/de.flapdoodle.embed.mongo#164]
Mongod does not cleanup extracted files after use.

The user.temp directory (c:\Users\...\AppData\Local\Temp\) is full with files like:
embedmongo-download-22057811-4f9d-4b54-a874-0af34340d33f.ZIP and extract-0040ced1-d389-4984-b1f3-811c0c95ffdamongod.exe .

Additionally every second start throws the following exception: 

> Caused by: java.io.IOException: Could not start process: ERROR: listen(): bind() failed errno:10048 Only one usage of each socket address (protocol/network address/port) is normally permitted. for socket: 127.0.0.1:12345

For me that means that the process is still running also.

Machine:
Windows (same issue on Centos Linux) 

Version:
de.flapdoodle.embed.mongo:1.50.5
springBootVersion = '1.3.0.RELEASE'
mongo.db.version=2.6.11

Spring Boot configuration:

```   @Bean(destroyMethod="close")
    public MongoClient mongo() throws IOException {
        LOG.info("Initializing embedded MongoDB instance");
        Logger logger = LoggerFactory.getLogger(MongodProcess.class);

        final String bindIp = env.getRequiredProperty("mongo.db.bindIp");
        final Integer port = env.getRequiredProperty("mongo.db.port", Integer.class);
        final String version = env.getRequiredProperty("mongo.db.version");   
        
        final String proxyHost = env.getRequiredProperty("mongo.db.proxyHost");
        final int proxyPort = env.getRequiredProperty("mongo.db.proxyPort", Integer.class);
        
        MongodStarter runtime = MongodStarter.getInstance(new RuntimeConfigBuilder()
                .defaults(Command.MongoD)
                .processOutput(new ProcessOutput(
                        new Slf4jStreamProcessor(logger, Slf4jLevel.TRACE),
                        new Slf4jStreamProcessor(logger, Slf4jLevel.WARN),
                        new Slf4jStreamProcessor(logger, Slf4jLevel.INFO)))
                .artifactStore(new ExtractedArtifactStoreBuilder()
                        .defaults(Command.MongoD)
                        //.executableNaming(new UserTempNaming())
                        .download(new DownloadConfigBuilder()
                                .defaultsForCommand(Command.MongoD)
                                .progressListener(new Slf4jProgressListener(logger))
                                .proxyFactory(new HttpProxyFactory(proxyHost, proxyPort))
                                .build())
                        .build())
                .build());
        MongodExecutable mongodExe = runtime.prepare(new MongodConfigBuilder()
                .version(parseVersion(version))
                .net(new Net(bindIp, port, Network.localhostIsIPv6()))
                .build());

        LOG.info("Starting embedded MongoDB instance");
        mongodExe.start();

        return new MongoClient(bindIp, port);
    } 
```

logs upon shutting down:

> o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [initandlisten] connection accepted from 127.0.0.1:52145 #1 (1 connection now open)
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] terminating, shutdown command received
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] dbexit: shutdown called
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] shutdown: going to close listening sockets...
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] closing listening socket: 476
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] shutdown: going to flush diaglog...
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] shutdown: going to close sockets...
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] shutdown: waiting for fs preallocator...
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.914+0200 [conn1] shutdown: closing all files...
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.915+0200 [conn1] closeAllFiles() finished
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.915+0200 [conn1] shutdown: removing fs lock...
o.s.b.a.mongo.embedded.EmbeddedMongo  2016-06-21T09:25:06.916+0200 [conn1] dbexit: really exiting now

The time I wait before starting it again is irrelevant

...and I also get the Windows firewal alert.
If I uncomment the 'executableNaming' section then: 

> Caused by: java.nio.file.FileAlreadyExistsException: C:\Users\...\AppData\Local\Temp\extract-...-extractmongod.exe

**Comment by bence-takacs** [comment]
An explicit call of 'mongodExe.stop();' resolved the issue

```
    @PreDestroy
	public void destroy() {
    	mongodExe.stop();
    }
```

The popup still persist. May I open a new issue for that?
